sklearn/decomposition/ksvd.py

```diff
@@ -0,0 +1,343 @@
+"""K-SVD dictionary learning algorithm.
+"""
+
+
+import numpy
+import scipy.sparse.linalg
+
+from ..base import BaseEstimator
+from .dict_learning import sparse_encode, SparseCodingMixin
+from ..preprocessing import normalize
+from ..utils import check_array, check_random_state
+from ..utils.extmath import norm
+from ..utils.random import choice
+
+
+# atoms with norm below are considered zero
+_ATOM_NORM_TOLERANCE = 1e-6
+
+
+def _worst_represented_example(X, dictionary, sparse_codes):
+    """Find the sample that has the most representation error."""
+
+    residuals = X - numpy.dot(sparse_codes, dictionary)
+    errors_squared = numpy.sum(residuals * residuals, axis=1)
+    worst_index = numpy.argmax(errors_squared)
+    return X[worst_index, :]
+
+
+def _decompose_svd(residual):
+    """Decompose the residual using 1-rank SVD."""
+
+    # Sparse version is used because non-sparse does not allow to
+    # set the desired number of singular values.
+    U, S, V = scipy.sparse.linalg.svds(residual, 1)
```

Comment:
Seems so. It also look like you can provide an initial vector approximation, could you try with present atom ?